ccw.core/src/java/ccw/CCWPlugin.java

```diff
@@ -109,31 +105,18 @@ public static ITracer getTracer() {
 			return NullTracer.INSTANCE;
 	}
     
-    public synchronized void startREPLServer() {
-    	if (ackREPLServer == null) {
-	        try {
-	        	// TODO use ClojureOSGi.withBundle instead
-	        	Var startServer = BundleUtils.requireAndGetVar(getBundle().getSymbolicName(), "clojure.tools.nrepl.server/start-server");
-	        	Object defaultHandler = BundleUtils.requireAndGetVar(
-	        	        getBundle().getSymbolicName(),
-	        	        "clojure.tools.nrepl.server/default-handler").invoke();
-	        	Object handler = BundleUtils.requireAndGetVar(
-	        	        getBundle().getSymbolicName(),
-	        	        "clojure.tools.nrepl.ack/handle-ack").invoke(defaultHandler);
-	            ackREPLServer = (ServerSocket)((Map)startServer.invoke(Keyword.intern("handler"), handler)).get(Keyword.intern("server-socket"));
-	            CCWPlugin.log("Started ccw nREPL server: nrepl://127.0.0.1:" + ackREPLServer.getLocalPort());
-	        } catch (Exception e) {
-	            CCWPlugin.logError("Could not start plugin-hosted REPL server", e);
-	            throw new RuntimeException("Could not start plugin-hosted REPL server", e);
-	        }
+    public void startREPLServer() {
+    	try {
+    	    ClojureInvoker.newInvoker(this, "ccw.core.launch")._("ccw-nrepl-start-if-necessary");
+    	} catch (Exception e) {
+    		CCWPlugin.logError("Could not start plugin-hosted REPL server", e);
+    		throw new RuntimeException("Could not start plugin-hosted REPL server", e);
     	}
     }
     
     public int getREPLServerPort() {
-    	if (ackREPLServer == null) {
-    		startREPLServer();
-    	}
-    	return ackREPLServer.getLocalPort();
+    	startREPLServer();
+    	return (Integer) ClojureInvoker.newInvoker(this, "ccw.core.launch")._("ccw-nrepl-port");
```

Comment:
No, but let's just do that.

I intend to merge the PR. Do you have any argument against bumping
tools.nrepl version from 0.2.5 to 0.2.9 ?

2015-03-25 10:32 GMT+01:00 Andrea Richiardi <notifications@github.com>:

> In ccw.core/src/java/ccw/CCWPlugin.java
> <https://github.com/laurentpetit/ccw/pull/36#discussion_r27105103>:
>
> >      	}
> >      }
> >
> >      public int getREPLServerPort() {
> > -    	if (ackREPLServer == null) {
> > -    		startREPLServer();
> > -    	}
> > -    	return ackREPLServer.getLocalPort();
> > +    	startREPLServer();
> > +    	return (Integer) ClojureInvoker.newInvoker(this, "ccw.core.launch")._("ccw-nrepl-port");
>
> Yes, this is why I was asking if the calls are cached...if there is a
> better way I am all ears :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laurentpetit/ccw/pull/36/files#r27105103>.
>



-- 
Laurent Petit